engine/src/contract_methods/connector.rs

```diff
@@ -404,3 +404,51 @@ pub fn set_paused_flags<I: IO + Copy, E: Env>(io: I, env: &E) -> Result<(), Cont
         Ok(())
     })
 }
+
+#[named]
+pub fn set_erc20_metadata<I: IO + Copy, E: Env, H: PromiseHandler>(
+    io: I,
+    env: &E,
+    handler: &mut H,
+) -> Result<SubmitResult, ContractError> {
+    with_hashchain(io, env, function_name!(), |io| {
+        let state = state::get_state(&io)?;
+        require_running(&state)?;
+        // TODO: Define special role for this transaction. Potentially via multisig?
```

Comment:
I proposed to have another role for this and grant this role to some internal DAO with participants from engine/bridge teams as a long term solution, and use the account `metadata.factory.bridge.near` as a short term.
But I'm not sure that this should be done in the scope of this PR, because looks like the roles system is not implemented in the engine, so this task can be more compacted than I expect.

Reply:
Also a while ago I proposed to retrieve the metadata from the nep141 using the cross contract call, but there were objections on this from Kirill to this because in some potential scenarios we can set a metadata that differ from the nep141.

Reply:
I think the right set of scenarios and permissions should be the following:
* Special role is able to obtain the metadata from the NEP-141 contract and set it.
* DAO can set any metadata if needed (for the rare special cases).